src/Tabs/Tab.js

```diff
@@ -47,22 +48,31 @@ export const styleSheet = createStyleSheet('MuiTab', (theme) => {
     fullWidth: {
       flex: '1 0 0',
     },
+    labelContainer: {
+      paddingTop: 6,
+      paddingBottom: 6,
+      paddingLeft: 12,
+      paddingRight: 12,
+      [theme.breakpoints.up('md')]: {
+        paddingLeft: 24,
```

Comment:
My initial instinct on this was that trying to define spacing in this manner would be very confusing to the consumer of Material-UI.  However, I was initially just thinking of this in a "global theme" manner.  After I started thinking about it as a component-level override, it started to make more sense.

In general, when it comes to Material Design, I believe in strictly following the specification, so overriding things like spacing doesn't make much sense to me.  However, I recognize that many people will want to do so, so allowing we either allow them to do it via theme or we allow them to do it via style overrides.

Presumably, most of the time they would want components to behave consistently wherever they happen to be used.  I'd then argue that theming makes much more sense than style overrides that have to be consistent applied.  So, based on that, I see a solid use case for a themed spacing override at the component level.

I did waffle back and forth quite a bit on my answer - I had actually crafted several reasons why I didn't think it worked - so I'm probably in the "not sold on the issue" camp as well.  However, for now, my opinion is coming down on using the theming and I've adjusted my PR accordingly.